sdks/python/apache_beam/io/gcp/bigtable_io.py

```diff
@@ -0,0 +1,120 @@
+manageable chunks and processed in parallel. In the example below
+we created a list of rows then passed to the GeneratedDirectRows
+DoFn to set the Cells and then we call the WriteToBigtable to insert
+those generated rows in the table.
+
+  main_table = (p
+       | 'Generate Direct Rows' >> GenerateDirectRows(number)
+       | 'Write to BT' >> beam.ParDo(WriteToBigtable(config)))
+"""
+from __future__ import absolute_import
+
+import apache_beam as beam
+from apache_beam.metrics import Metrics
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.transforms.display import DisplayDataItem
+
+try:
+  from google.cloud.bigtable import Client
+  from google.cloud.bigtable.batcher import MutationsBatcher
+except ImportError:
+  pass
+
+
+class WriteToBigtable(beam.DoFn):
+  """ Creates the connector can call and add_row to the batcher using each
+  row in beam pipe line
+
+  :type beam_options: class:`~bigtable_configuration.BigtableConfiguration`
+  :param beam_options: class `~bigtable_configuration.BigtableConfiguration`
+  """
+
+  def __init__(self, beam_options):
+    super(WriteToBigtable, self).__init__(beam_options)
+    self.beam_options = beam_options
+    self.table = None
+    self.batcher = None
+    self.written = Metrics.counter(self.__class__, 'Written Row')
+
+  def start_bundle(self):
+    if self.table is None:
+      client = Client(project=self.beam_options.project_id)
+      instance = client.instance(self.beam_options.instance_id)
+      self.table = instance.table(self.beam_options.table_id)
+    self.batcher = MutationsBatcher(self.table)
+
+  def process(self, row):
+    self.written.inc()
+    self.batcher.mutate(row)
+
+  def finish_bundle(self):
+    result = self.batcher.flush()
+    self.batcher = None
+    return result
+
+  def display_data(self):
+    return {'projectId': DisplayDataItem(self.beam_options.project_id,
+                                         label='Bigtable Project Id'),
+            'instanceId': DisplayDataItem(self.beam_options.instance_id,
+                                          label='Bigtable Instance Id'),
+            'tableId': DisplayDataItem(self.beam_options.table_id,
+                                       label='Bigtable Table Id')
+           }
+
+
+class BigtableConfiguration(object):
+  """ Bigtable configuration variables.
+
+  :type project_id: :class:`str` or :func:`unicode <unicode>`
+  :param project_id: (Optional) The ID of the project which owns the
+    instances, tables and data. If not provided, will
+    attempt to determine from the environment.
+
+  :type instance_id: str
+  :param instance_id: The ID of the instance.
+
+  :type table_id: str
+  :param table_id: The ID of the table.
+  """
+
+  def __init__(self, project_id, instance_id, table_id):
+    self.project_id = project_id
+    self.instance_id = instance_id
+    self.table_id = table_id
+
+
+class WriteBigtableOptions(PipelineOptions):
```

Comment:
Where is this used ?

Reply:
Right now, We going to set that class as a PipelineOptions, but, It is not necessary, so, I will delete it, because, We only need the project_id, instance_id and table_id in the pipeline.